mesonbuild/interpreter/interpreter.py

```diff
@@ -2758,12 +2758,24 @@ def extract_incdirs(self, kwargs, key: str = 'include_directories'):
         return result
 
     @typed_pos_args('include_directories', varargs=str)
-    @typed_kwargs('include_directories', KwargInfo('is_system', bool, default=False))
+    @typed_kwargs(
+        'include_directories',
+        KwargInfo('is_system', bool, default=False),
+        KwargInfo(
+            'use_type',
+            str,
+            default='both',
+            since='1.2.0',
+            validator=in_set_validator(set(e.value for e in build.IncludeDirsUse)),
+            convertor=lambda x: next(e for e in build.IncludeDirsUse if x == e.value)
+        ),
+    )
     def func_include_directories(self, node: mparser.BaseNode, args: T.Tuple[T.List[str]],
                                  kwargs: 'kwtypes.FuncIncludeDirectories') -> build.IncludeDirs:
```

Comment:
You're right. It was missing, but its absence caused no problems.  Still, I've now added it.
Is the intention that those kwargs.py classes get used just for type checking?  Based on that use, would you have expected that mypy should have picked up my omission from FuncIncludeDirectories?

Reply:
> Is the intention that those kwargs.py classes get used just for type checking?

Yes.

> Based on that use, would you have expected that mypy should have picked up my omission from FuncIncludeDirectories?

No, because interpreter.py is incomplete and thus ends up excluded from coverage.